packages/cli/src/program.ts

```diff
@@ -42,51 +33,126 @@ let programInstance: Command | null = null;
+      if (option.parser) {
+        programInstance.option(
+          option.flags,
+          option.description,
+          option.parser as (value: string, previous: unknown) => unknown,
+          option.defaultValue as string | boolean | number,
+        );
+      }
+      else {
+        programInstance.option(
+          option.flags,
+          option.description,
+          option.defaultValue as string | boolean | string[],
+        );
+      }
+    }
+
+    // Unified preAction hook: handles config resolution, then logging/reporting setup
+    programInstance.hook('preAction', async (thisCommand, actionCommand) => {
+      const targetCommand = actionCommand ?? thisCommand;
+
+      // Step 1: Resolve and apply configuration
+      const ancestry = getCommandAncestry(targetCommand);
+      const resolvedConfig = await resolveConfig(targetCommand, ancestry);
+      applyConfigToCommander(ancestry, resolvedConfig);
+      setActiveConfig(resolvedConfig);
+      logActiveConfig(resolvedConfig, ancestry, resolvedConfig.verbose);
+
+      // Step 2: Setup logging, UI, and reporting with resolved config
+      const options = targetCommand.optsWithGlobals();
+      const commandPieces: string[] = [];
+      for (
+        let c: Command | null = targetCommand;
+        c;
+        c = c.parent as Command | null
+      ) {
+        commandPieces.unshift(c.name());
+      }
+      const command = commandPieces.join(' ');
+
+      const runId = Date.now();
+
+      // Initialize logger with transports based on config
+      const transports: LogTransport[] = [];
+
+      // Add console transport if enabled
+      if (resolvedConfig.log.console.enabled) {
+        transports.push(
+          new ConsoleTransport({
+            level: resolvedConfig.log.console.level as LogLevel,
+          }),
+        );
+      }
+
+      // Add file transport if enabled
+      if (resolvedConfig.log.file.enabled) {
+        const logsPath = resolveCommandPath(
+          directories.log,
+          options.space,
+          options.path,
+        );
         const logFilename = `${commandPieces.join('-')}-${runId}.jsonl`;
         const filePath = path.join(logsPath, logFilename);
-        transports.push(new FileTransport({
-          filePath,
-          maxFiles: 10,
-        }));
-        getLogger({
-          context: { runId, command, options, cliVersion: packageJson.version },
-          transports,
-        });
+        transports.push(
+          new FileTransport({
+            filePath,
+            level: resolvedConfig.log.file.level as LogLevel,
+            maxFiles: resolvedConfig.log.file.maxFiles,
+          }),
+        );
+      }
+
+      // Initialize logger with configured transports
+      getLogger({
+        context: { runId, command, options, cliVersion: packageJson.version },
+        transports,
+      });
 
-        getUI({ enabled: true });
+      // Initialize UI
+      getUI({ enabled: true });
```

Comment:
issue: We should make this configurable so console logging can be enabled without the UI polluting the console.